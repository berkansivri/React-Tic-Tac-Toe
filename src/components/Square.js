import React, { useContext } from 'react'
import GameContext from '../context/game-context'

const Square = ({ location, sign }) => {
  const { dispatch, turn, setTurn } = useContext(GameContext)

  const handleOnClick = () => {
    if(!!sign) return
    dispatch({ type: "ADD_MOVE", sign: turn, location })
    setTurn(turn === "X" ? "O" : "X")
  }
  
  return (
    <button className="square" onClick={handleOnClick}>
      {sign}
    </button>
  )
}

export { Square as default }
