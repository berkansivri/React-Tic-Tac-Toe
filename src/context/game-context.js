import React, { useContext } from 'react'

const gameContext = React.createContext()

const useGameContext = () => useContext(gameContext)

export { gameContext as default, useGameContext }
