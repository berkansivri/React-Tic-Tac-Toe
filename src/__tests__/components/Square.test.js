import React from 'react'
import { shallow } from 'enzyme'
import Square from '../../components/Square'
import * as Context from '../../context/game-context'
import GameContext from '../../context/game-context'

let contextValues, SquareComp

beforeEach(() => {
  contextValues = {
    dispatch: jest.fn(),
    turn: "X",
    setTurn: jest.fn()
  }
  SquareComp = () => (
    <GameContext.Provider value={{...contextValues}}>
      <Square location={1} sign={"X"} />
    </GameContext.Provider>
  )
})

test("should render square component correctly", () => {
  const wrapper = shallow(<SquareComp />)
  expect(wrapper).toMatchSnapshot()
})

test("handle on sign square", () => {
  jest.spyOn(Context, 'useGameContext')
      .mockImplementation(() => contextValues)
  
  const wrapper = shallow(<Square location={1}/>)
  wrapper.find("button").simulate("click")
  const { dispatch, turn, setTurn } = contextValues
  expect(dispatch).toHaveBeenLastCalledWith({ type:"ADD_MOVE", sign: turn, location: 1 })
  expect(setTurn).toHaveBeenLastCalledWith("O")
})