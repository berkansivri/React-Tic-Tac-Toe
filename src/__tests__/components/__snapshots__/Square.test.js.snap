// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render square component correctly 1`] = `
<ContextProvider
  value={
    Object {
      "dispatch": [MockFunction],
      "setTurn": [MockFunction],
      "turn": "X",
    }
  }
>
  <Square
    location={1}
    sign="X"
  />
</ContextProvider>
`;
